platform_umbrella/apps/common_core/lib/common_core/resources/resource_generator/virtual_service_builder.ex

```diff
@@ -22,6 +22,20 @@ defmodule CommonCore.Resources.VirtualServiceBuilder do
     add_route(virtual_service, route)
   end
 
+  @spec maybe_https_redirect(VirtualService.t(), boolean()) :: VirtualService.t()
+  def maybe_https_redirect(%VirtualService{} = virtual_service, false = _ssl_enabled?), do: virtual_service
+
+  def maybe_https_redirect(%VirtualService{} = virtual_service, true = _ssl_enabled?) do
+    {:ok, route} =
+      HTTPRoute.new(
+        name: "https-redirect",
+        match: [%{uri: %{prefix: "/"}, scheme: %{exact: "http"}}],
+        redirect: %{scheme: "https"}
+      )
+
+    prepend_route(virtual_service, route)
+  end
```

Comment:
Please add some documentation explaining what's going on.  As I understand it, we're creating a route match that will redirect to https for the prefix of /. Since the well known acme paths are added later and are more specific this acts as a catch all?